torch/testing/_internal/common_distributed.py

```diff
@@ -247,8 +247,33 @@ def _run(cls, rank, test_name, file_name):
     def _join_processes(self, fn):
         timeout = get_timeout(self.id())
         start_time = time.time()
-        for p in self.processes:
-            p.join(timeout)
+        subprocess_error = False
+        while True:
```

Comment:
This is a busy loop? do we need to insert some short sleeps in it?

Reply:
Yes, I guess we should insert some sleeps. Although if our sleep interval is too long we could lose some benefit of being able to return quickly in the case of errors. Ideally this process would use a pipe or condvar of some sort that just sleeps until timeout or until one process exits.

Reply:
If you would like to land this sooner, a 10ms sleep should also be fine. We already have this in our test. Totally up to you :)

https://github.com/pytorch/pytorch/blob/24dd800e6a1471e595099dc9fdd48c19c71a4e20/torch/testing/_internal/distributed/rpc/dist_autograd_test.py#L51-L53

https://github.com/pytorch/pytorch/blob/24dd800e6a1471e595099dc9fdd48c19c71a4e20/torch/testing/_internal/distributed/rpc/dist_autograd_test.py#L1366-L1368

https://github.com/pytorch/pytorch/blob/24dd800e6a1471e595099dc9fdd48c19c71a4e20/torch/testing/_internal/distributed/rpc/rpc_test.py#L1413-L1421